ts/Stock/Indicators/OBV/OBVIndicator.ts

```
@@ -0,0 +1,244 @@
+            yVal: Array<number> | Array<Array<number>> = (series.yData as any),
+            OBV: Array<Array<number>> = [],
+            xData: Array<number> = [],
+            yData: Array<number> = [];
+
+        let OBVPoint: Array<number> = [],
+            i: number = 0,
+            previousOBV: number = 0,
+            curentOBV: number = 0,
+            previousClose: number = 0,
+            curentClose: number = 0,
+            volume: Array<number>,
+            closeValues: Array<number>,
+            trend: number;
+
+        // Checks if volume series exists.
+        if (volumeSeries) {
+            closeValues = this.getCloseValues(yVal);
+            volume = ((volumeSeries as Series).yData as any);
+
+            for (i; i < closeValues.length; i++) {
+                // Add first point and qet close value.
+                if (i === 0) {
+                    OBVPoint = [xVal[i], previousOBV];
+                    previousClose = closeValues[i];
+                } else {
+                    curentClose = closeValues[i];
+                    trend = this.getTrend(curentClose, previousClose);
+
+                    if (trend === 1) {
+                        curentOBV = previousOBV + volume[i];
+                    }
+                    if (trend === 0) {
+                        curentOBV = previousOBV;
+                    }
+                    if (trend === -1) {
+                        curentOBV = previousOBV - volume[i];
+                    }
+
+                    // Add point.
+                    OBVPoint = [xVal[i], curentOBV];
+
+                    // Asing currend as previous for next iteration
+                    previousOBV = curentOBV;
+                    previousClose = curentClose;
+                }
+
+                OBV.push(OBVPoint);
+                xData.push(xVal[i]);
+                yData.push(OBVPoint[1]);
+            }
+        } else {
+            return;
```

Could you throw an error when volume series was not found? See other indicators (A/D, MFI etc)